python/tests/integration/arcticdb/version_store/test_basic_version_store.py

```diff
@@ -107,6 +141,51 @@ def test_special_chars(s3_version_store, special_char):
     assert_frame_equal(vitem.data, df)
 
 
+@pytest.mark.parametrize("breaking_char", ["&", "*", "<", ">"])
+def test_s3_breaking_chars(s3_version_store, breaking_char):
+    """Test that chars that are not supported are raising the appropriate exception and that we fail on write without corrupting the db
+    """
+    sym = f"prefix{breaking_char}postfix"
+    df = sample_dataframe()
+    with pytest.raises(ArcticNativeNotYetImplemented):
+        s3_version_store.write(sym, df)
+
+    assert sym not in s3_version_store.list_symbols()
+
+
+@pytest.mark.parametrize("unhandled_char", [chr(0), chr(127), chr(128)])
+def test_unhandled_chars_default(s3_version_store, unhandled_char):
+    """Test that by default, the problematic chars are raising an exception"""
+    sym = f"prefix{unhandled_char}postfix"
+    df = sample_dataframe()
+    with pytest.raises(UserInputException):
+        s3_version_store.write(sym, df)
+    syms = s3_version_store.list_symbols()
+    assert sym not in syms
+
+
+@pytest.mark.parametrize("unhandled_char", [chr(0)])
+def test_unhandled_chars_no_strict_check(s3_version_store, unhandled_char):
+    """Test that when we turn the STRICT_SYMBOL_CHECK off, the problematic \x00 is raising an exception"""
+    set_config_int("VersionStore.NoStrictSymbolCheck", 1)
+    sym = f"prefix{unhandled_char}postfix"
+    df = sample_dataframe()
+    s3_version_store.write(sym, df)
+    with pytest.raises(InternalException):
+        s3_version_store.list_symbols()
+    unset_config_int("VersionStore.NoStrictSymbolCheck")
```

Comment:
There's an `arcticdb.util.test.config_context` which does this in an exception-safe way